Tests/Expected/XCAssets/swift4-context-all-customname.swift

```diff
@@ -0,0 +1,133 @@
+// Generated using SwiftGen, by O.Halligon — https://github.com/SwiftGen/SwiftGen
+
+#if os(OSX)
+  import AppKit.NSImage
+  typealias Color = NSColor
+  typealias Image = NSImage
+#elseif os(iOS) || os(tvOS) || os(watchOS)
+  import UIKit.UIImage
+  typealias Color = UIColor
+  typealias Image = UIImage
+#endif
+
+// swiftlint:disable file_length
+
+@available(*, deprecated, renamed: "XCTImageAsset")
+typealias XCTAssetsType = XCTImageAsset
+
+struct XCTImageAsset {
+  fileprivate var value: String
+
+  var image: Image {
+    let bundle = Bundle(for: BundleToken.self)
+    #if os(iOS) || os(tvOS)
+    let image = Image(named: value, in: bundle, compatibleWith: nil)
+    #elseif os(OSX)
+    let image = bundle.image(forResource: NSImage.Name(value))
+    #elseif os(watchOS)
+    let image = Image(named: value)
+    #endif
+    guard let result = image else { fatalError("Unable to load image \(value).") }
+    return result
+  }
+}
+
+struct XCTColorAsset {
+  fileprivate var value: String
```

Comment:
That way if we rename `value` to `name` to be more consistent, we could imagine later add properties conveying the color components as found in the JSON of the asset catalog (probably useless so maybe no need to add them for now, but we could add some variables for them in the context for people wanting to use that Parser to generate other outputs like HTML color palettes from the parsed asset catalog or whatnot…